sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/signaling/TrouterUtils.java

```diff
@@ -158,6 +165,25 @@ public static ChatMessageType parseChatMessageType(String rawType) {
         return ChatMessageType.TEXT;
     }
 
+    public static Map<String, String> parseChatMessageMetadata(String rawMetadata) {
+        Map<String, String> metadata = Collections.<String, String>emptyMap();
+
+        if (rawMetadata == null || TextUtils.isEmpty(rawMetadata)) {
+            return metadata;
+        }
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            metadata = mapper.readValue(rawMetadata, new TypeReference<HashMap<String, String>>() { });
+        } catch (JsonMappingException e) {
+            CLIENT_LOGGER.error(e.getMessage());
+        } catch (JsonProcessingException e) {
+            CLIENT_LOGGER.error(e.getMessage());
+        }
```

Comment:
Consider adding a catch all branch here otherwise it is harder for caller to handle error. Optinally :-)